sycl/test/usm/math.cpp

```diff
@@ -0,0 +1,81 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+
+#include <array>
+#include <cassert>
+#include <cmath>
+
+namespace s = cl::sycl;
+
+int main() {
+  s::queue myQueue;
+  // fract with unified shared memory
+  {
+    s::cl_float r{ 0 };
+    s::cl_float i{ 999 };
+    {
+      s::cl_float *Buf = (s::cl_float *)s::malloc_shared(sizeof(cl_float) * 2,
+          myQueue.get_device(), myQueue.get_context());
+      s::malloc_shared(100, myQueue.get_device(), myQueue.get_context());
+      myQueue.submit([&](s::handler &cgh) {
+        cgh.single_task<class fractF1UF1>([=]() {
+          Buf[0] = s::fract(s::cl_float{ 1.5f }, &Buf[1]);
```

Comment:
Can you also test these built-in' functions with vector type?

Reply:
Added. The code didn't work at first, so I had to do some more modification to get it work. Now I really wish I could use constexpr if...